improver/cli/gradient_between_vertical_levels.py

```diff
@@ -0,0 +1,42 @@
+# (C) Crown copyright, Met Office. All rights reserved.
+#
+# This file is part of IMPROVER and is released under a BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""CLI to calculate a gradient between two vertical levels."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(*cubes: cli.inputcubelist):
+    """
+    Calculate the gradient between two vertical levels. The gradient is calculated as the
+    difference between the input cubes divided by the difference in height.
+
+    Input cubes can be provided at height or pressure levels. If the cubes are provided
+    at a pressure level, the height above sea level is extracted from height_of_pressure_levels
+    cube. If the cubes are provided at height levels this is assumed to be a height above ground
+    level and the height above sea level is calculated by adding the height of the orography to
+    the "height" coordinate of the cube. It is possible for one cube to be defined at height
+    levels and the other at pressure levels.
+
+    Args:
+        cubes (iris.cube.CubeList):
+            Contains two cubes of a diagnostic at two different vertical levels. The cubes must
+            either have a height coordinate or a pressure coordinate. If either cube is defined at
+            height levels, an orography cube must also be provided. If either cube is defined at
```

Comment:
While this all works, the case where both cubes contain height levels doesn't really need an orography cube, as the orography height gets cancelled out ((A + orog) - (B + orog) == (A - B)). 

If we chose to allow this, I think the way to implement it would be to add a boolean `height_is_agl = not(orography or geopotential_height)`, and then use this to raise an exception if a pressure cube is found and to omit the addition of orography to height cubes.

Should we? It would be a little more efficient for the "both_height" case. Do we have any of these cases yet?

Reply:
Currently we don't have a use case for this but I was trying to make this plugin as generic as possible so I've tried to add it. I was over-thinking how to do this but I think what I've ended up implementing works as intended.